index.js

```diff
@@ -41,7 +68,19 @@ app.get("/movies/:Title", (req, res) => {
 
 //Get list of movies by director
 app.get("/movies/Director/:Director", (req, res) => {
-  Movies.findOne({Director: req.params.Director})
+  Movies.find({'Director.Name': req.params.Director})
+  .then((movie) => {
+  res.json(movie);
+  })
+  .catch((err) => {
+  console.error(err);
+  res.status(500).send('error: ' + err);
+  });
+});
+
+//Get list of movies by genre
+app.get("/movies/Genre/:Genre", (req, res) => {
+  Movies.find({'Genre.Name': req.params.Genre})
   .then((movie) => {
   res.json(movie);
```

Comment:
This endpoint is similar to the one the fetch movies by Director as it returns a list of movies by Genre.